elastic/beats
# Missing backup and delete for AWS S3 polling mode

## Proposed commit message



When running Filebeat s3 input polling mode, we suppose to have the options to backup s3 objects into a specific bucket and delete the object after backup.
```
filebeat.inputs:
- type: aws-s3
  enabled: true
  credential_profile_name: elastic-observability
  bucket_arn: arn:aws:s3:::test-s3-bucket-ks
  default_region: eu-central-1
  backup_to_bucket_arn: arn:aws:s3:::test-s3-bucket-ks-backup
  delete_after_backup: true
  number_of_workers: 5
```
With this config, objects/logs will be ingested from the bucket specified in bucket_arn into Elasticsearch just fine but no backup or delete happened.

This PR is add the feature back into S3 polling mode when user specifies a backup bucket and/or delete after backup.

## Checklist




## How to test this PR locally

With the same config, after adding this change, rebuild filebeat and we can see the new log `log7` got ingested into Elasticsearch and after ingestion, this file is deleted from the `test-s3-bucket-ks` bucket and copied to `test-s3-bucket-ks-backup` bucket.
<img width="1604" alt="Screenshot 2024-10-01 at 4 44 42 PM" src="https://github.com/user-attachments/assets/31151024-c664-415d-9351-3925b354e226">

## Related issues

- Closes https://github.com/elastic/beats/issues/41070
